Run the diff script also in generate_reference_trees.sh

pieterdavid opened the pull request:
It is supposed to "fail" (show differences), but will show what changes are about to be pushed.
This should save a step (and prevent common errors - maybe we should even add `scram b`?)

pieterdavid committed:
Run the diff script also in generate_reference_trees.sh

It is supposed to "fail" (show differences), but will show what changes
are about to be pushed

OlivierBondu commented:
excellent idea :)

as for the `scram b`, I'm guessing crab in no way knows about the hash of the packages it compiled last ? This would be the ideal: suggest to build only if there are changes (bref, `make` as it should be)

(also, looks like I tagged a bit too soon, sorry about that, but well it shouldn't matter for running a production...)

pieterdavid committed:
Merge branch 'CMSSW_8_0_6p' into diff_in_updateref

pieterdavid committed:
More checks (git, scram) in generate_reference_trees

Added:
- check that the CMSSW project area is clean/committed
- check that the Framework directory is clean/committed
- compile everything
before remaking the reference trees.
The checks can be disabled with --skipgitcheck and --skipscram;
see also test/generate_reference_trees.sh --help

pieterdavid committed:
update README

OlivierBondu commented:
Thanks a lot!!
